pkg/collector/parser/receiver.go

```diff
@@ -77,8 +77,45 @@ func IsRegistered(name string) bool {
 var (
 	endpointKey      = "endpoint"
 	listenAddressKey = "listen_address"
+	scraperReceivers = map[string]struct{}{
```

Comment:
Should we not include `prometheus` in this list too?

Reply:
There is no top-level `endpoint` for Prometheus config. However, it is covered with a note just below the lines of this change. I don't mind moving it to this list.

Reply:
yeah i think we should move it to this list given it provides the same functionality as the case in the switch below.

Reply:
Updated